pyteal/ast/scratchvar_test.py

```diff
@@ -104,3 +117,101 @@ def test_scratchvar_assign_load():
     actual = TealBlock.NormalizeBlocks(actual)
 
     assert actual == expected
+
+
+def test_dynamic_scratchvar_type():
+    myvar_default = DynamicScratchVar()
+    assert myvar_default.storage_type() == TealType.anytype
+    assert myvar_default.store(Bytes("value")).type_of() == TealType.none
+    assert myvar_default.load().type_of() == TealType.anytype
+
+    with pytest.raises(TealTypeError):
+        myvar_default.store(Pop(Int(1)))
+
+    myvar_int = ScratchVar(TealType.uint64)
+    assert myvar_int.storage_type() == TealType.uint64
+    assert myvar_int.store(Int(1)).type_of() == TealType.none
+    assert myvar_int.load().type_of() == TealType.uint64
+
+    with pytest.raises(TealTypeError):
+        myvar_int.store(Bytes("value"))
+
+    with pytest.raises(TealTypeError):
+        myvar_int.store(Pop(Int(1)))
+
+    myvar_bytes = ScratchVar(TealType.bytes)
```

Comment:
`myvar_int` and `myvar_bytes` aren't `DynamicScratchVar`s

Reply:
👍